File: backend/app/schemas/analytics.py
from pydantic import BaseModel
from typing import Dict, Any, List, Optional

class MetricCategory(BaseModel):
    enabled: bool
    name: str
    description: str
    metrics: Dict[str, Any]
    charts: Optional[List[Dict[str, Any]]] = None

class SMSMetrics(BaseModel):
    delivery_metrics: MetricCategory
    engagement_metrics: Optional[MetricCategory]
    quality_compliance: MetricCategory
    cost_efficiency: MetricCategory
    performance_metrics: MetricCategory
    business_impact: MetricCategory
    ai_specific: MetricCategory
    security_fraud: MetricCategory

class EmailMetrics(BaseModel):
    pass  # To be implemented

class VoiceMetrics(BaseModel):
    pass  # To be implemented

class Overview(BaseModel):
    totalInteractions: int
    totalCost: float
    averageResponseTime: float
    successRate: float

class DateRange(BaseModel):
    start: str
    end: str

class AnalyticsData(BaseModel):
    sms: SMSMetrics
    email: Dict[str, Any]  # To be implemented
    voice: Dict[str, Any]  # To be implemented
    overview: Overview
    dateRange: DateRange

File: frontend/src/types/analytics.ts
export interface MetricCategory {
  enabled: boolean;
  name: string;
  description: string;
  metrics: {
    [key: string]: number | string;
  };
  charts?: {
    type: 'line' | 'bar' | 'pie';
    data: any[];
  }[];
}

export interface Message {
  id: string;
  content: string;
  timestamp: string;
  direction: 'inbound' | 'outbound';
  status: string;
  phoneNumber: string;
}

export interface Conversation {
  id: string;
  messages: Message[];
  phoneNumber: string;
  lastMessageAt: string;
  status: string;
}

export interface SMSMetrics {
  totalMessages: number;
  inboundMessages: number;
  outboundMessages: number;
  averageResponseTime: number;
  deliveryRate: number;
  conversations: Conversation[];
  hourlyActivity: { hour: number; count: number }[];
}

export interface VoiceMetrics {
  totalCalls: number;
  inboundCalls: number;
  outboundCalls: number;
  averageDuration: number;
  successRate: number;
  hourlyActivity: { hour: number; count: number }[];
}

export interface EmailMetrics {
  totalCount: number;
  openRate: number;
  clickRate: number;
  bounceRate: number;
  hourlyActivity: { hour: number; count: number }[];
}

export interface AnalyticsData {
  smsMetrics: SMSMetrics;
  voiceMetrics: VoiceMetrics;
  emailMetrics: EmailMetrics;
  totalInteractions: number;
  customerSatisfaction: number;
  responseTime: number;
  resolutionRate: number;
}
